### PR TITLE
dev-libs/device-atlas-api-c: Bump to 2.1.2_1 version

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

**Comment by devnexen** [comment]
Any opinion ? :-) since the previous version of this library is not available anymore it might be nice to update it :-) thanks.

File: dev-libs/device-atlas-api-c/device-atlas-api-c-2.1.2.ebuild [hunk 1]
```diff
@@ -1,38 +1,44 @@
-# Copyright 1999-2016 Gentoo Foundation
+# Copyright 1999-2017 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=6
-inherit unpacker cmake-multilib
+inherit cmake-multilib
 
-MY_P="deviceatlas-enterprise-c-${PV}"
+MY_P="deviceatlas-enterprise-c-${PV}_1"
```
Comment (mgorny) [inline dev-libs/device-atlas-api-c/device-atlas-api-c-2.1.2.ebuild]: Are you sure that upstream will never release _2? If not, then I think it'd be better to name it `2.1.2_p1` and then `${PV/_p/_}`.

File: dev-libs/device-atlas-api-c/device-atlas-api-c-2.1.2.ebuild [hunk 2]
```diff
@@ -1,38 +1,44 @@
...
 	${RDEPEND}"
 
 RESTRICT="fetch mirror bindist"
 
 S="${WORKDIR}/${MY_P}"
 
-PATCHES=(
-	"${FILESDIR}/${PV}-cmake-install.patch"
-)
+#PATCHES=(
```
Comment (mgorny) [inline dev-libs/device-atlas-api-c/device-atlas-api-c-2.1.2.ebuild]: Why is this commented out?

File: dev-libs/device-atlas-api-c/device-atlas-api-c-2.1.2.ebuild [hunk 3]
```diff
@@ -1,38 +1,44 @@
...
+#	"${FILESDIR}/${PV}-src-cmakelists.patch"
+#)
 
 pkg_nofetch() {
 	eerror "Please go to https://deviceatlas.com/deviceatlas-haproxy-module"
 	eerror "And download DeviceAtlas C API"
 	eerror "Then place the file in ${DISTDIR}/${MY_P}.zip"
 }
 
+src_unpack() {
```
Comment (mgorny) [inline dev-libs/device-atlas-api-c/device-atlas-api-c-2.1.2.ebuild]: Don't put patches in src_unpack. It's code from EAPI 0, no longer valid with EAPI 2+.

File: dev-libs/device-atlas-api-c/device-atlas-api-c-2.1.2.ebuild [hunk 4]
```diff
@@ -1,38 +1,44 @@
...
+	unpack ${A}
+	cd "${S}"
+	epatch "${FILESDIR}/${PV}-src-cmakelists.patch"
```
Comment (mgorny) [inline dev-libs/device-atlas-api-c/device-atlas-api-c-2.1.2.ebuild]: Do not use epatch in EAPi 6. There's `eapply` instead (which is used by PATCHES).

File: dev-libs/device-atlas-api-c/files/2.1.2-src-cmakelists.patch [hunk 5]
```diff
@@ -0,0 +1,10 @@
+--- Src/CMakeLists.txt.orig	2017-03-24 19:43:41.237879211 +0000
```
Comment (mgorny) [inline dev-libs/device-atlas-api-c/files/2.1.2-src-cmakelists.patch]: Adjust the patch to apply with -p1, i.e. add `a/`, `b/` path prefixes.

File: dev-libs/device-atlas-api-c/files/2.1.2-src-cmakelists.patch [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
++++ Src/CMakeLists.txt	2017-03-24 19:44:00.598878508 +0000
+@@ -37,5 +37,6 @@
+ set (CMAKE_SHARED_LINKER_FLAGS -m32)
+ endif()
+  
+-install(FILES libda.so libda.so.${API_VERSION} DESTINATION /usr/local/lib)
++install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libda.so
++	${CMAKE_CURRENT_BINARY_DIR}/libda.so.${API_VERSION} DESTINATION /usr/local/lib)
```
Comment (mgorny) [inline dev-libs/device-atlas-api-c/files/2.1.2-src-cmakelists.patch]: I'm pretty sure this is wrong. We do not want anything installed in /usr/local.

File: dev-libs/device-atlas-api-c/device-atlas-api-c-2.1.2_p1.ebuild [hunk 7]
```diff
@@ -33,6 +33,12 @@ pkg_nofetch() {
 	eerror "Then place the file in ${DISTDIR}/${MY_P}.zip"
 }
 
+src_unpack() {
```
Comment (mgorny) [inline dev-libs/device-atlas-api-c/device-atlas-api-c-2.1.2_p1.ebuild]: This is still wrong and entirely unnecessary. The default implementation (which happens when you do not declare anything) should work here just fine.

File: dev-libs/device-atlas-api-c/files/2.1.2-src-cmakelists.patch [hunk 8]
```diff
@@ -0,0 +1,11 @@
+--- a/Src/CMakeLists.txt.orig	2017-03-24 19:43:41.237879211 +0000
++++ b/Src/CMakeLists.txt	2017-03-24 19:44:00.598878508 +0000
+@@ -37,5 +37,6 @@
+ set (CMAKE_SHARED_LINKER_FLAGS -m32)
+ endif()
+  
+-install(FILES libda.so libda.so.${API_VERSION} DESTINATION /usr/local/lib)
++install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libda.so
++	${CMAKE_CURRENT_BINARY_DIR}/libda.so.${API_VERSION} DESTINATION /usr/lib)
```
Comment (mgorny) [inline dev-libs/device-atlas-api-c/files/2.1.2-src-cmakelists.patch]: This does not account for multilib libdirs. I don't really understand why you don't reuse GNUInstallDirs as the old patch did.

File: dev-libs/device-atlas-api-c/device-atlas-api-c-2.1.ebuild [hunk 9]
```diff
@@ -12,7 +12,7 @@ SRC_URI="${MY_P}.zip"
 
 LICENSE="all-rights-reserved"
 SLOT="0"
-KEYWORDS="amd64 ~arm ppc x86"
```
Comment (mgorny) [inline dev-libs/device-atlas-api-c/device-atlas-api-c-2.1.ebuild]: You have added the outdated version. But I'll fix it for you.

**Pull request closed without merging**